app/main.py

```diff
@@ -112,6 +112,8 @@ def handle_suspend():
         cordon_all_nodes(k8s_v1, protect_removal_disabled, exclude_node_id=hibernation_node_id)
         time.sleep(20)
     else:
+        logging.info("No suitable hibernation node found, terminating hibernation, resuming autoscaler")
```

Comment:
If create `create_hibernation_node`  fails I think we will not reach this line. `hibernation_node_id ` will be none and we will try to get node_name and later status from not existing node. From customer provided logs we can see las fail was `[create_hibernation_node]: Failed to get Operation status` https://castai.atlassian.net/browse/CSU-522?focusedCommentId=20847 

Maybe we could do something like  this:

```
try:
 handle_suspend()
except:
  handle_resume()
  ```

Reply:
Another option would be to use `restartPolicy: OnFailure` and do a check at the start of a job - if autoscaler is disabled: run resume and terminate job - previous job failed to do something and failed

If autoscaler is ON run suspend.

Yes, this runs on the assumption that autoscaler will not be disabled by the user - current resume and suspend also ignore this so maybe it is not a big problem.

I actually like this approach more since we are guarded against everything